zeroeq/event.cpp

```diff
@@ -9,19 +9,20 @@
 namespace zeroeq
 {
 
-Event::Event( const uint128_t& type )
-    : _impl( new detail::Event( type ))
-{}
-
-Event::Event( Event&& rhs )
-    : _impl( rhs._impl )
+Event::Event( const uint128_t& type, const EventFunc& func )
+    : _impl( new detail::Event( type, func ))
 {
-    rhs._impl = 0;
+    if( func)
+        setUpdatedFunction( [this](){ _impl->func( *this ); });
 }
 
 Event::~Event()
 {
-    delete _impl;
+}
+
+std::string Event::getTypeName() const
+{
+    return _impl->type.getString();
```

Comment:
This is not the fully qualified, demangled class name. Also it's not clear to me how the JSON code gets its schema to do the serialization. Probably we should forbid REST/JSON for FlatBuffers.